OPTM-63 Only show QR scanner button when platform is web, android, iOS.

guisedlythe opened the pull request:
- When QR scanner is not supported for the platform, only allow adding devices with the "device form screen"

OPTM-63 #close

guisedlythe committed:
OPTM-63 Only show QR scanner button when platform is web, android, iOS.

- When QR scanner is not supported for the platform, only allow adding devices with the "device form screen"

OPTM-63 #close

guisedlythe commented:
I think I forgot to create a PR for this (minor) change

JacobRoe commented on lib/screens/devices/device_list_screen.dart:
```diff
@@ -40,39 +41,29 @@ class _DeviceListScreenState extends State<DeviceListScreen> with LogMixin {
-        peerService.callbackProvider,
-      ),
+    final showQrScannerButton = _showQrScannerButton();
+    final consumerWidget = Consumer2<PeerInfoService, PeerService>(
```
Shouldn't it be `ChangeCallbackNotifier<PeerInfoService>` and `ChangeCallbackNotifier<PeerService>`?

JacobRoe commented on lib/screens/devices/device_list_screen.dart:
```diff
@@ -98,8 +90,11 @@ class _DeviceListScreenState extends State<DeviceListScreen> with LogMixin {
-                Text('Longpress the button below to manually add a device.'),
+                showQrScannerButton
+                    ? Text('Press the button below to scan a QR code.')
+                    : Text('Press the button below to add a device'),
```
Maybe add a period to the sentence to keep it consistent.

JacobRoe reviewed (changes requested):
LGTM except the things I commented on.

guisedlythe committed:
OPTM-63 Changes according to PR review.

- Added '.' at the end of the display text
- Fixed usage of old way to get provider

JacobRoe reviewed (approved):
Thank you. Looks good.
